ui/pages/confirmations/components/confirm/blockaid-loading-indicator/blockaid-loading-indicator.test.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import configureStore from 'redux-mock-store';
+import {
+  TransactionStatus,
+  TransactionType,
+} from '@metamask/transaction-controller';
+
+import mockState from '../../../../../../test/data/mock-state.json';
+import { BlockaidResultType } from '../../../../../../shared/constants/security-provider';
+import { renderWithProvider } from '../../../../../../test/lib/render-helpers';
+import { SecurityAlertResponse } from '../../../types/confirm';
+
+import BlockaidLoadingIndicator from './blockaid-loading-indicator';
+
+const mockSecurityAlertResponse: SecurityAlertResponse = {
+  securityAlertId: 'test-id-mock',
+  reason: 'test-reason',
+  result_type: BlockaidResultType.Loading,
+};
+
+const render = (
+  securityAlertResponse: SecurityAlertResponse = mockSecurityAlertResponse,
+) => {
+  const currentConfirmationMock = {
+    id: '1',
+    status: TransactionStatus.unapproved,
+    time: new Date().getTime(),
+    type: TransactionType.personalSign,
+    securityAlertResponse,
+    chainId: '0x1',
+  };
+
+  const mockExpectedState = {
+    ...mockState,
+    metamask: {
+      ...mockState.metamask,
+      unapprovedPersonalMsgs: {
+        '1': { ...currentConfirmationMock, msgParams: {} },
+      },
+      pendingApprovals: {
+        '1': {
+          ...currentConfirmationMock,
+          origin: 'origin',
+          requestData: {},
+          requestState: null,
+          expectsResult: false,
+        },
+      },
+      preferences: { redesignedConfirmationsEnabled: true },
+      signatureSecurityAlertResponses: {
+        'test-id-mock': securityAlertResponse,
+      },
+    },
+    confirm: { currentConfirmation: currentConfirmationMock },
+  };
+
+  const defaultStore = configureStore()(mockExpectedState);
+  return renderWithProvider(<BlockaidLoadingIndicator />, defaultStore);
+};
+
+describe('BlockaidLoadingIndicator', () => {
+  it('returns spinner when there blockaid validation is in progress for signature', () => {
+    const { container } = render();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('returns null if blockaid validaiton is not in progress', () => {
```

Comment:
Minor, but would we benefit from a test to ensure it renders correctly with no Blockaid response at all?

Reply:
PR is updated to address this.